### ISSUE TITLE
Error when downloading a file from SharePoint Drive - "contentType is required"

### ISSUE THREAD

**Issue by SKumarSpace** [microsoftgraph/msgraph-sdk-go#205]
Hello,

I'm trying to download a file from a SharePoint Document Library (Drive).
But when I do so, I am getting the error: "`contentType is required`".
When I run the request using the `NewDriveItemItemRequestBuilder`, I get the same error.

However, when running the request manually through Postman, it works successfully.
Any ideas as to what might be causing this error.

Minimal reproduction below:

```golang
package main

import (
	"fmt"
	"log"

	azidentity "github.com/Azure/azure-sdk-for-go/sdk/azidentity"
	a "github.com/microsoft/kiota-authentication-azure-go"
	msgraphsdk "github.com/microsoftgraph/msgraph-sdk-go"
	content "github.com/microsoftgraph/msgraph-sdk-go/drive/items/item"
)

func checkError(err error) {
	if err != nil {
		log.Fatalln(err)
	}
}

func main() {
	CLIENT_ID := "client-id"
	CLIENT_SECRET := "client-secret"
	TENANT_ID := "tenant-id"

	cred, err := azidentity.NewClientSecretCredential(TENANT_ID, CLIENT_ID, CLIENT_SECRET, nil)

	checkError(err)

	auth, err := a.NewAzureIdentityAuthenticationProviderWithScopes(cred, []string{fmt.Sprintf("https://graph.microsoft.com/.default")})
	checkError(err)

	adapter, err := msgraphsdk.NewGraphRequestAdapter(auth)
	checkError(err)

	client := msgraphsdk.NewGraphServiceClient(adapter)

	driveId := "drive-id"
	itemId := "item-id"

	// Method A:
	fileBytes, err := client.DrivesById(driveId).ItemsById(itemId).Content().Get()
	checkError(err)

	fmt.Printf("len(fileBytes) = %d\n", len(fileBytes))

	// Method B:
	targetUrl := fmt.Sprintf("https://graph.microsoft.com/v1.0/drives/%s/items/%s/content", driveId, itemId)
	requestBuilder := content.NewDriveItemItemRequestBuilder(targetUrl, adapter)
	resp, err := requestBuilder.Get()
	checkError(err)

	fmt.Printf("len(fileBytes) = %d\n", len(resp.GetContent()))
}

```

**Comment by baywet** [comment]
Hi @SKumarSpace,
Thanks for trying the Go SDK and for reporting this issue.
This is a bug in the generator, which [I've addressed](https://github.com/microsoft/kiota/pull/1708), once merged, the fix will go out with the next weekly refresh.

Long story short the [request adapter was expecting](https://github.com/microsoft/kiota-http-go/blob/e5034f26ec8fdf9eaf9a9ffa2828864f9a7ff191/nethttp_request_adapter.go#L343) `[]byte` by the [generator providing](https://github.com/microsoftgraph/msgraph-sdk-go/blob/e26ee4114d4b449b8e2f387b2dee47d2bacc930c/drives/item/items/item/content/content_request_builder.go#L97) `byte`

**Comment by baywet** [comment]
Hi @SKumarSpace,
Thanks for your patience on this one.
#211 fixed that issue and was just released a couple of minutes ago as v0.29.0
As you can see the operation now has []byte, which the request adapter will map to the byte stream from the response.
https://github.com/microsoftgraph/msgraph-sdk-go/blob/32e95517554481b1846846df8198fdab79909f76/drives/item/items/item/content/content_request_builder.go#L97

Let us know if you need anything else.
